### PR TITLE
Improvements: ProcessService returns user processes

### PR DESCRIPTION
**Pull request by lciolecki**

### CODE REVIEW

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/process/DBProcessServiceSpec.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import slick.jdbc.{HsqldbProfile, JdbcBackend}
+
+import java.time.Duration
+
+class DBProcessServiceSpec extends FlatSpec with Matchers with PatientScalaFutures {
+  import org.scalatest.prop.TableDrivenPropertyChecks._
+  import pl.touk.nussknacker.ui.api.helpers.TestCategories._
+  import pl.touk.nussknacker.ui.api.helpers.TestProcessUtil._
+
+  import scala.concurrent.ExecutionContext.Implicits.global
```
Comment (mproch) [inline ui/server/src/test/scala/pl/touk/nussknacker/ui/process/DBProcessServiceSpec.scala:23]: can we move those to the top-level imports?

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/process/DBProcessServiceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  private val processCategoryService = new ConfigProcessCategoryService(ConfigWithScalaVersion.config)
+  private val newProcessPreparer = TestFactory.createNewProcessPreparer()
+
+  private val dummyDbConfig: Config = ConfigFactory.parseString("""db {url: "jdbc:hsqldb:mem:none"}""".stripMargin)
+  private val dummyDb: DbConfig = DbConfig(JdbcBackend.Database.forConfig("db", dummyDbConfig), HsqldbProfile)
+
+  private val DefaultRequestTimeLimit = Duration.ofMinutes(1)
+
+  it should "return processes for each user" in {
+    val dummyWriteProcessRepository = TestFactory.newWriteProcessRepository(dummyDb)
```
Comment (mproch) [inline ui/server/src/test/scala/pl/touk/nussknacker/ui/process/DBProcessServiceSpec.scala:49]: can we move those declarations outside of this method? They seem generic test setup

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala [hunk 3]
```diff
@@ -235,6 +236,12 @@ class DBProcessService(managerActor: ActorRef,
         Future(Left(ProcessNotFoundError(processIdWithName.id.value.toString)))
     }
 
+  override def getUserProcesses[PS: ProcessShapeFetchStrategy](user: LoggedUser): Future[List[BaseProcessDetails[PS]]] = {
+    val userCategories = processCategoryService.getUserCategories(user)
+    val shapeStrategy = implicitly[ProcessShapeFetchStrategy[PS]]
```
Comment (mproch) [inline ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala:241]: no quite sure why it's better than `implicit val luser: LoggedUser: user` ;)

Comment (lciolecki) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala:241]: It looks more scala pro! :) But to be honest I don't say it's better but I think it looks better.

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/process/DBProcessServiceSpec.scala (hunk 2 above)
Comment (lciolecki) [reply ui/server/src/test/scala/pl/touk/nussknacker/ui/process/DBProcessServiceSpec.scala:49]: Moved

File: ui/server/src/test/scala/pl/touk/nussknacker/ui/process/DBProcessServiceSpec.scala (hunk 1 above)
Comment (lciolecki) [reply ui/server/src/test/scala/pl/touk/nussknacker/ui/process/DBProcessServiceSpec.scala:23]: Moved

File: ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala [hunk 4]
```diff
@@ -67,6 +67,7 @@ trait ProcessService {
 
   def updateProcess(processIdWithName: ProcessIdWithName, action: UpdateProcessCommand)(implicit user: LoggedUser): Future[XError[UpdateProcessResponse]]
 
+  def getUserProcesses[PS: ProcessShapeFetchStrategy](user: LoggedUser): Future[List[BaseProcessDetails[PS]]]
```
[the remark this answers is not in the document]
Comment (lciolecki) [reply ui/server/src/main/scala/pl/touk/nussknacker/ui/process/ProcessService.scala:70]: Hmm, ok you're right ;)

**Review by mproch — APPROVED** [review]

**Pull request merged**